### PR TITLE
Searchbar

### PR DESCRIPTION
**Pull request by danielmolnar**
Implemented search bar

### CODE REVIEW

File: client/src/Store.js [hunk 1]
```diff
@@ -13,3 +13,8 @@ const Store = ({ children }) => {
 };
 
 export default Store;
+
+// value={{
+//   checkWatchlist: [checkWatchlist, setCheckWatchlist],
+//   search: [search, setSearch],
+// }}
```
Comment (miriamhaenle) [inline client/src/Store.js:20]: Remove unused code before merging to main

File: client/src/components/AddButton.js [hunk 2]
```diff
@@ -1,14 +1,14 @@
 import React, { useContext } from 'react';
 import PropTypes from 'prop-types';
 import styled from 'styled-components';
-import { Context } from '../hooks/Store';
+import { Context } from '../Store';
```
Comment (miriamhaenle) [inline client/src/components/AddButton.js:4]: As you could potentially use multiple Context's you might consider renaming this one to convey which context this is about. Something like `WatchlistContext`maybe?

File: client/src/components/CardInfos.js [hunk 3]
```diff
@@ -1,43 +1,44 @@
...
-  addToWatchList,
-}) {
+export default function CardInfos({ open, onClose, addToWatchList, movie }) {
   if (!open) return null;
+  const baseUrl = 'https://image.tmdb.org/t/p/original/';
+  const release = movie.first_air_date || movie.release_date;
+  let check;
+  movie.poster_path === null && movie.backdrop_path === null
+    ? (check = true)
+    : (check = false);
```
Comment (miriamhaenle) [inline client/src/components/CardInfos.js:16]: This was hard for me to understand. 
Probably the following will be sufficient as the check on the right will either return true or false thus your check will hold one of each.  
`const check = movie.poster_path === null && movie.backdrop_path === null`

You could also reconsider the name - what does it check for?

File: client/src/components/Poster.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import styled, { css } from 'styled-components';
+import PropTypes from 'prop-types';
+import flixbuddies_poster from '../images/flixbuddies_poster.png';
+
+function Poster({ isLarge, movie }) {
+  const baseUrl = 'https://image.tmdb.org/t/p/original/';
+  let check;
+  movie.poster_path === null && movie.backdrop_path === null
+    ? (check = true)
+    : (check = false);
```
Comment (miriamhaenle) [inline client/src/components/Poster.js:11]: Same here as in CardInfos.js

Comment (miriamhaenle) [reply client/src/components/Poster.js:11]: Maybe a good name could be `pathExisting` or something in this direction

File: client/src/components/Searchbar.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+export default function Searchbar({ getQuery }) {
+  const [text, setText] = useState('');
+  const onChange = (q) => {
+    setText(q);
+    getQuery(q);
+  };
+
+  return (
+    <>
```
Comment (miriamhaenle) [inline client/src/components/Searchbar.js:13]: No need for the fragments

File: client/src/components/Searchbar.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+export default function Searchbar({ getQuery }) {
+  const [text, setText] = useState('');
+  const onChange = (q) => {
```
Comment (miriamhaenle) [inline client/src/components/Searchbar.js:7]: What does `q`stand for? `query`? Probably an unnecessary abbreviation, no need to save characters :)

**Review by miriamhaenle — CHANGES_REQUESTED** [review]
👏  Very nice! I just added a few suggestions :)

File: client/src/App.js [hunk 7]
```diff
@@ -1,26 +1,45 @@
 import { Switch, Route } from 'react-router-dom';
+import React, { useState, useEffect } from 'react';
```
[the remark this answers is not in the document]
Comment (danielmolnar) [reply client/src/App.js:2]: Right!

File: client/src/Pages/Search.js [hunk 8]
```diff
@@ -1,5 +1,53 @@
...
+  }
+
+  function toggleWatchList(movie) {
+    setCheckWatchlist(isOnWatchList(movie));
+  }
+
+  return isLoading ? (
+    <>
+      <Spinner />
+    </>
```
[the remark this answers is not in the document]
Comment (danielmolnar) [reply client/src/Pages/Search.js:29]: True, code still working :)

File: client/src/components/AddButton.js (hunk 2 above)
Comment (danielmolnar) [reply client/src/components/AddButton.js:4]: Not yet planning to use more Context, but still you are right :)

File: client/src/components/CardInfos.js (hunk 3 above)
Comment (danielmolnar) [reply client/src/components/CardInfos.js:16]: Funny, Thomas made the very same suggestions just a couple of hours ago :) The line provides a boolean so I can handle the background/poster with own pictures whenever the API fails to provide data. Good point about changing the name :)

File: client/src/App.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (danielmolnar) [reply client/src/App.js:2]: Thanks a lot for your review!

**Pull request merged**